meilisearch/templates/pvc.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if .Values.persistence.enabled }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ include "meilisearch.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "meilisearch.name" . }}
+    helm.sh/chart: {{ include "meilisearch.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  accessModes:
+    - {{ .Values.persistence.accessMode | quote }}
+  resources:
+    requests:
+      storage: {{ .Values.persistence.size | quote }}
+{{- if .Values.persistence.storageClass }}
+{{- if (eq "-" .Values.persistence.storageClass) }}
+  storageClassName: ""
+{{- else }}
+  storageClassName: "{{ .Values.persistence.storageClass }}"
+{{- end }}
```

Comment:
We would still need to support setting the `storageClassName` variable to empty, since the k8s spec supports it. You can read in more detail [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1), but the essence is for the storage class value is:
* If it has a value other than `""`, it will only attach to PV which are created from that storageClass
* If the value is `""`, it will only attach to a PVs with no class (no annotation or one set equal to "" in the PV definition)
* If the value is not set, the PVC attachment process will vary if the default storage class plugin is enabled or not

Reply:
The default should be to not specify it in the `values.yaml` file. I'll change that

Reply:
When I install the chart in a DigitalOcean Kubernetes, and I enable persistence in `values.yaml` (leaving everything else untouched) the installation works but the statesfulset shows this error:

`running "VolumeBinding" filter plugin for pod "meilisearch-0": pod has unbound immediate PersistentVolumeClaims `

Reply:
I am going to deploy a Kubernetes cluster in DO to troubleshoot then. It may be related to your cluster configuration where if you have a default storage class, then all dynamic PVs are being created associated to that storageclass and therefore the PVC won't bind to any of them.

If `storageClassName: ""`, it requires a PV that has been manually created in advanced

Reply:
I have created a do cluster and there is a storageclass set to be the default one. When I ran:
```
k get storageclass
```
I got the following output:
```
NAME                         PROVISIONER                 RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
do-block-storage (default)   dobs.csi.digitalocean.com   Delete          Immediate           true                   46m
```

So, setting the `storageClassName: ""` would require to already have PersistentVolument provisioned and available to be bound by the PVC that the meilisearch chart creates.

With the last change that I pushed a couple of days ago, just setting the `persistence.enabled: true` works because now it won't add the `storageClassName` field at all and a PV will be created automatically bye leveraging the default storageclass

Reply:
Thanks, this is perfect, It works fine for me too! 👌